docs/android/linter.md

```diff
@@ -0,0 +1,150 @@
+:::
+
+## Why use a linter?
+
+Using a linter ensures:
+
+- **Consistency**: Enforces a standard code style, similar to our [codestyle](codestyle).
+- **Focus**: Allows reviewers to focus on logic rather than formatting or trivial issues.
+- **Prevention**: Helps avoid crashes and bugs by catching common mistakes, such as using APIs not supported by the target Android API level.
+
+For example, failing to check the Android API version before using an unsupported API can lead to crashes.
+
+## Linters used in the project
+
+### KTLint
+
+We use [KTLint](https://pinterest.github.io/ktlint) as our Kotlin linter, integrated via [Gradle plugin](https://github.com/JLLeitschuh/ktlint-gradle). The configuration is located in the main `build.gradle.kts` file. We mostly use the default configuration but enable [SARIF](tips/sarif_reports) reports for GitHub Actions to annotate issues in pull requests.
+
+#### Ignoring an issue
+
+Always try to fix issues rather than ignoring them. If ignoring is necessary, follow these steps:
+
+1. Use the `@Suppress` annotation for specific constructs:
+   ```kotlin
+   @Suppress("trailing-comma-on-call-site")
+    fun myCallSiteExample() {
+        myFunction(
+            "value1",
+            "value2", // This trailing comma would normally cause a warning
+        )
+    }
+   ```
+
+2. For project-wide suppression, update the `.editorconfig` file as per [this guide](https://pinterest.github.io/ktlint/0.49.1/faq/#how-do-i-globally-disable-a-rule-without-editorconfig). Open a dedicated PR with an explanation for disabling the rule:
+    ```editorconfig
+    ...
+    # Allow trailing commas but do not enfoce it to follow Kotlin convention
+    ktlint_standard_trailing-comma-on-call-site = disabled
+    ij_kotlin_allow_trailing_comma_on_call_site = true
+    ktlint_standard_trailing-comma-on-declaration-site = disabled
+    ij_kotlin_allow_trailing_comma = true
+    ```
+
+#### Running KTLint locally
+
+Run the following command to check all code in the repository:
+
+```bash
+./gradlew ktlintCheck :build-logic:convention:ktlintCheck --continue
+```
+
+:::note
+Use `--continue` to get all issues across Gradle modules instead of stopping at the first failure.
+:::
+
+You can add this check to be run automatically through a git [pre-commit hook](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks) by running this command
+
+```bash
+./gradlew addKtlintCheckGitPreCommitHook
+```
+
+### Android Linter
+
+The Android linter is enabled for all variants to ensure comprehensive checks. Its configuration is located in `build-logic/convention/src/main/kotlin/AndroidCommonConventionPlugin.kt`. SARIF reports are generated for GitHub Actions to annotate issues in pull requests.
+
+#### Ignoring an issue
+
+Follow these steps to ignore an issue:
+
+1. Use the `@Suppress` annotation for specific constructs.
+2. Add the issue to the `lint-baseline.xml` file.
```

Comment:
Do we need a "see below" comment here? Hopefully when its my time to update the baseline I remember where to look lol